### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by mohashyne**
# New Features added

### This pull request introduces several key functionalities to enhance user interaction within the application. Here's a breakdown of the changes:

    

- [ ] Current User Method:
        Added a method named current_user to the ApplicationController. This method provides access to the current user's data across all controllers.
        The current_user method retrieves the first user from the database for the current session. Note that proper authentication will be implemented in future projects to enhance security.

    

- [ ] Post and Comment Functionality:
        Created forms to facilitate the following actions:
            Create Post: Users can now generate posts on the platform. The post is attributed to the current user using the current_user method from the ApplicationController.
            Create Comment: Users have the ability to comment on posts. The comment is associated with the current user through the current_user method.

- [ ] Post Likes:
        Implemented the functionality to allow users to add likes to posts. This feature enhances user engagement and interaction within the application.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,26 +1,39 @@
...
-
-    # Rest of the logic to show the posts on the view
+  def create
+    @post = @user.posts.build(post_params)
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post was successfully created.'
+    else
+      flash.now[:alert] = 'Post creation failed!'
+      render :new
+    end
```
Comment (anagudelogu) [inline app/controllers/posts_controller.rb]: - [x] According to the requirements:

image

Please consider using the `current_user` method that you created earlier to create a new post! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @mohashyne, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (mohashyne) [reply app/controllers/posts_controller.rb]: update post controller to create a post on behalf of  current_user

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @mohashyne , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**